sky/serve/replica_managers.py

```diff
@@ -904,12 +966,27 @@ def _replica_prober(self) -> None:
             logger.debug('Running replica prober.')
             try:
                 self._probe_all_replicas()
-                replica_statuses = [
-                    info.status for info in serve_state.get_replica_infos(
-                        self._service_name)
-                ]
+                replica_infos = serve_state.get_replica_infos(
+                    self._service_name)
+                replica_statuses = [info.status for info in replica_infos]
                 serve_utils.set_service_status_from_replica_statuses(
                     self._service_name, replica_statuses)
+
+                # Clean old version
+                current_oldest_version = min([
+                    info.version for info in replica_infos
+                ]) if replica_infos else self.oldest_version
+                if self.oldest_version < current_oldest_version:
+                    for version in range(self.oldest_version,
+                                         current_oldest_version):
+                        task_yaml = self._get_yaml(version)
+                        # Delete old version metadata.
+                        serve_state.delete_version(self._service_name, version)
+                        # Delete storage buckets of older versions.
+                        service.cleanup_storage(task_yaml)
+                    # newest version will be cleaned in serve down
+                    self.oldest_version = current_oldest_version
```

Comment:
Could we do this in autoscaler scale down? Also, we need to cleanup this in `service.py::_start`.

Reply:
moved to `scale_down()` in `replica_managers`.
added clean up for versions in `service.py::_cleanup`.

Reply:
It feels strange to do such cleanup in `_replica_prober`; is there any other possible position to place it?

Reply:
Probably do this in refresh process pool; after a replica's termination

Reply:
sounds good. let me do it in _refresh_process_pool.